oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/reactive/function/client/ServerOAuth2AuthorizedClientExchangeFilterFunction.java

```diff
@@ -144,6 +187,31 @@ private static ReactiveOAuth2AuthorizedClientManager createDefaultAuthorizedClie
 		return authorizedClientManager;
 	}
 
+	/**
+	 * Creates the default {@link #responseHandler}, based on type of the given {@code authorizedClientManager}.
+	 *
+	 * <p>For well-known client manager implementations, error responses will be
+	 * handled by delegating to the client manager's authorization failure handler.</p>
+	 *
+	 * @param authorizedClientManager the {@link ReactiveOAuth2AuthorizedClientManager} which manages the authorized client(s)
+	 * @return the default response handler
+	 */
+	private BiFunction<ClientRequest, Mono<ClientResponse>, Mono<ClientResponse>> createDefaultResponseHandler(
+			ReactiveOAuth2AuthorizedClientManager authorizedClientManager) {
+		if (authorizedClientManager instanceof DefaultReactiveOAuth2AuthorizedClientManager) {
+			return new DefaultResponseHandler(
+					((DefaultReactiveOAuth2AuthorizedClientManager) authorizedClientManager)::getAuthorizationFailureHandler);
+		} else if (authorizedClientManager instanceof AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager) {
+			return new DefaultResponseHandler(
+					((AuthorizedClientServiceReactiveOAuth2AuthorizedClientManager) authorizedClientManager)::getAuthorizationFailureHandler);
+		} else if (authorizedClientManager instanceof UnAuthenticatedReactiveOAuth2AuthorizedClientManager) {
+			return new DefaultResponseHandler(
+					((UnAuthenticatedReactiveOAuth2AuthorizedClientManager) authorizedClientManager)::getAuthorizationFailureHandler);
+		} else {
+			return (request, response) -> response;
+		}
+	}
```

Comment:
I'm not terribly happy with this.  It's quite ugly.  Suggestions welcome.

Reply:
Resolved by not automatically supporting removal of authorized client on auth failure when this constructor is used.  Requires user to call setAuthorizationFailureHandler to enable this functionality.